### PR TITLE
Use autoreconf instead of bootstrap

### PR DESCRIPTION
**Pull request by vadz**
Please see commit message for the reasons.

Also update the README slightly as a side effect.

### CODE REVIEW

**Comment by vadz** [comment]
Sorry for the initial version which broke the build with the docs, but now it seems like it should work and I think it's preferable to avoid modifying the source directory when building in another one, so it's another (unplanned) improvement here.

Please let me know if you see any problems with doing it like this. TIA!

File: docs/Makefile.am [hunk 1]
```diff
@@ -1,5 +1,8 @@
 
-html_DATA = $(srcdir)/html/*.*
+html_DATA = $(builddir)/html/*.*
+
+$(html_DATA):
```
Comment (vslavik) [inline docs/Makefile.am:4]: Is this going to work? I'm not sure if the in-git version did (does Automake handle wildcards?), but I don't see how could it handle the wildcard in `builddir` correctly…

I *think* the goal here was to have `make install` install the docs, and that it won't anymore. That's probably why `bootstrap` built the docs too.

The HTML build certainly should be included in the dist tarball (which I think still works).

File: bootstrap [hunk 2]
```diff
@@ -1,76 +0,0 @@
...
-# Use --copy with automake to allow simultaneous use on windows under mingw and
-# cygwin platforms. Symlinking of files under mingw does not work out for
-# cygwin and vice-versa.
-echo "Setting up build system for xmlwrapp:"
-echo " - aclocal " && aclocal && \
-echo " - libtoolize " && $LIBTOOLIZE --copy --automake && \
-echo " - autoconf " && autoconf && \
-echo " - automake " && automake --add-missing --copy && \
-echo " - doxygen "  && run_optional_tool docs doxygen && \
-echo " - bakefile " && run_optional_tool platform/Win32 bkl xmlwrapp.bkl && \
```
Comment (vslavik) [inline bootstrap:70]: What about this bakefile bit? MSVS projects should be in the tarball, I think, so it's either running it as part of `make dist` or committing projects into the repository (after switching to `msvs`).

**Review by vslavik — APPROVED** [review]

File: docs/Makefile.am (hunk 1 above)
Comment (vadz) [reply docs/Makefile.am:4]: It seems to work, but I'm not really sure why/how myself, so I've changed the makefile to use a Doxygen tag file as a dependency, this seems to be simpler and more clear.

File: bootstrap (hunk 2 above)
Comment (vadz) [reply bootstrap:70]: Oops, thanks for noticing this. I'll add a `dist-hook` to run bkl for now and will also add `make dist` to the CI builds to ensure that it doesn't get broken again.

**Comment by vadz** [comment]
It has taken embarrassingly many tries, but `make dist` seems to work now.

File: scripts/travis/script.sh [hunk 3]
```diff
@@ -13,6 +13,30 @@ if [ -n "$HOST" ]; then
...
+
+    make dist
+
+    XMLWRAPP_VERSION=$(sed -n -e 's/AC_INIT(xmlwrapp, \([0-9.]*\),.*)$/\1/p' configure.ac)
+
+    # Same as above, we assume tar can uncompress .gz files on the fly.
+    tar xf xmlwrapp-${XMLWRAPP_VERSION}.tar.gz
+
+    cd xmlwrapp-${XMLWRAPP_VERSION}
+    ./configure CXXFLAGS="$CXXFLAGS" $configure_args
```
Comment (vslavik) [inline scripts/travis/script.sh:37]: This part can be replaced with a `make distcheck` one-liner, with the added bonus of being [extra unfriendly](https://www.gnu.org/software/automake/manual/html_node/Checking-the-Distribution.html) to the makefile and trying to trigger some of the commonly forgotten edge cases (make dist from the dist itself, builddir!=srcdir etc.)

**Comment by vslavik** [comment]
Thanks for the updates (and the cleanup in the first place)!

**Pull request merged**